### ISSUE TITLE
Vidcoder Beta 3.0 now uses QSV Decoder for 8 bit HEVC when doing QSV HEVC Encode. Less CPU % used. Thank you. Please add QSV decoding to non-QSV encoding.

### ISSUE THREAD

**Issue by ZaphodX77** [RandomEngy/VidCoder#274]
10 bit HEVC decode is a Handbrake issue, due to be fixed in 1.1.0.

    "HWDecode": false,
    "QSV": {
      "Decode": true,
      "AsyncDepth": 0
    }
  }
}
[07:34:41] starting job
[07:34:41] work: track 1, dithering not supported by codec
[07:34:41] work: only 1 chapter, disabling chapter markers
[07:34:41] job configuration:
[07:34:41]  * source
[07:34:41]    + C:\Temp1\8bit 900p X265.mkv
[07:34:41]    + title 1, chapter(s) 1 to 1
[07:34:41]    + container: matroska,webm
[07:34:41]  * destination
[07:34:41]    + C:\Temp1\8bit 900p X265 K29.mkv
[07:34:41]    + container: Matroska (libavformat)
[07:34:41]  * video track
[07:34:41]    + decoder: hevc_qsv

**Comment by ZaphodX77** [comment]
I should have shown the bottom of the previous log. It will terminate without an error in about 3 minutes.
Suddenly, this stopped working. It stops at a syncing as shown, but not for every test file:
[08:55:23]    + decoder: hevc_qsv
[08:55:23]    + filters
[08:55:23]      + Framerate Shaper (mode=1:rate=27000000/1126125)
[08:55:23]        + frame rate: 23.976 fps -> constant 23.976 fps
[08:55:23]      + Crop and Scale (width=1920:height=804:crop-top=2:crop-bottom=2:crop-left=0:crop-right=0)
[08:55:23]        + source: 1920 * 808, crop (2/2/0/0): 1920 * 804, scale: 1920 * 804
[08:55:23]    + Output geometry
[08:55:23]      + storage dimensions: 1920 x 804
[08:55:23]      + pixel aspect ratio: 1 : 1
[08:55:23]      + display dimensions: 1920 x 804
[08:55:23]    + encoder: H.265 (Intel Media SDK)
[08:55:23]      + preset:  balanced
[08:55:23]      + profile: main
[08:55:23]      + level:   4.0
[08:55:23]      + quality: 27.00 (QP)
[08:55:23]  * audio track 1
[08:55:23]    + decoder: English (Opus) (2.0 ch) (track 1, id 0x1)
[08:55:23]      + samplerate: 48000 Hz
[08:55:23]    + mixdown: Stereo
[08:55:23]    + gain: 1dB
[08:55:23]    + encoder: Opus (libopus)
[08:55:23]      + bitrate: 64 kbps, samplerate: 48000 Hz
[08:55:23]      + compression level: 10.00
[08:55:23] sync: expecting 192702 video frames
[08:55:23] encqsvInit: using encode-only path
[08:55:23] encqsvInit: H.265/HEVC Main profile @ level 4.0
[08:55:23] encqsvInit: TargetUsage 4 AsyncDepth 5
[08:55:23] encqsvInit: GopRefDist 4 GopPicSize 32 NumRefFrame 3
[08:55:23] encqsvInit: BFramesMax 3 BRefType pyramid
[08:55:23] encqsvInit: RateControlMethod CQP with QPI 27 QPP 29 QPB 31
[08:55:24] sync: first pts audio 0x1 is 0

If the input is HEVC 10bit, it encodes without QSV decoding. This only happens on QSV decoding, and it can happen even when the source file is X264. Still, a file I used first worked fine, but now it doesn't!!

This made me test adding a chapter, since this one worked, but that didn't fix that first file for QSV decoding since it still stopped encoding at the point above.
[09:03:46] encqsvInit: RateControlMethod CQP with QPI 27 QPP 29 QPB 31
[09:03:46] sync: first pts video is 0
[09:03:46] sync: "Chapter 1" (1) at frame 1 time 0
[09:03:46] sync: first pts audio 0x1 is 0
[09:03:46] qsv_enc_init: using 'hardware (1) via D3D11' implementation, API: 1.25

So, basically Megusta, RMTeam, PSA & YST files encode just fine (mostly), using QSV decode to encode HEVC. Was on Intel 4815 driver. Going to install 4849 now & see what happens.

**Comment by ZaphodX77** [comment]
I was wrong.  The beta does not decode_QSV reliably, except for those file download brands I mention above.

Now with Intel 4849 driver. This is what happens, and is probably what happened before, 1st test that I thought worked, on this problem file. I thought it was encoding before because the FPS lingers from the previous encode in the bottom of Vidcoder. It was not advancing in %, however, and the encode "completes" showing no error, Green Check, Succeeded, with an unplayable file of small size.

  "Video": {
    "Encoder": 512,
    "Level": "4.0",
    "TwoPass": false,
    "Turbo": false,
    "ColorMatrixCode": 0,
    "Options": "",
    "Preset": "balanced",
    "Profile": "main",
    "Quality": 32.0,
    "HWDecode": false,
    "QSV": {
      "Decode": true,
      "AsyncDepth": 0
    }
  }
}
[09:55:35] starting job
[09:55:35] work: track 1, dithering not supported by codec
[09:55:35] work: only 1 chapter, disabling chapter markers
[09:55:35] job configuration:
[09:55:35]  * source
[09:55:35]    + C:\Temp1\900p.mkv
[09:55:35]    + title 1, chapter(s) 1 to 1
[09:55:35]    + container: matroska,webm
[09:55:35]  * destination
[09:55:35]    + C:\Temp1\900p X265 K32.mkv
[09:55:35]    + container: Matroska (libavformat)
[09:55:35]  * video track
[09:55:35]    + decoder: hevc_qsv
[09:55:35]    + filters
[09:55:35]      + Framerate Shaper (mode=1:rate=27000000/1126125)
[09:55:35]        + frame rate: 23.976 fps -> constant 23.976 fps
[09:55:35]      + Crop and Scale (width=1200:height=900:crop-top=0:crop-bottom=0:crop-left=0:crop-right=0)
[09:55:35]        + source: 1200 * 900, crop (0/0/0/0): 1200 * 900, scale: 1200 * 900
[09:55:35]    + Output geometry
[09:55:35]      + storage dimensions: 1200 x 900
[09:55:35]      + pixel aspect ratio: 1 : 1
[09:55:35]      + display dimensions: 1200 x 900
[09:55:35]    + encoder: H.265 (Intel Media SDK)
[09:55:35]      + preset:  balanced
[09:55:35]      + profile: main
[09:55:35]      + level:   4.0
[09:55:35]      + quality: 32.00 (QP)
[09:55:35]  * subtitle track 1, English [UTF-8] (track 0, id 0x2, Text) -> Passthrough, Default
[09:55:35]  * audio track 1
[09:55:35]    + decoder: English (AAC) (5.1 ch) (track 1, id 0x1)
[09:55:35]      + samplerate: 48000 Hz
[09:55:35]    + mixdown: Dolby Surround
[09:55:35]    + gain: 1dB
[09:55:35]    + encoder: Opus (libopus)
[09:55:35]      + bitrate: 64 kbps, samplerate: 48000 Hz
[09:55:35]      + compression level: 10.00
[09:55:35] sync: expecting 91424 video frames
[09:55:35] encqsvInit: using encode-only path
[09:55:35] encqsvInit: H.265/HEVC Main profile @ level 4.0
[09:55:35] encqsvInit: TargetUsage 4 AsyncDepth 5
[09:55:35] encqsvInit: GopRefDist 4 GopPicSize 32 NumRefFrame 3
[09:55:35] encqsvInit: BFramesMax 3 BRefType pyramid
[09:55:35] encqsvInit: RateControlMethod CQP with QPI 32 QPP 34 QPB 36
[09:55:35] sync: first pts audio 0x1 is 0
[09:55:36] sync: first pts subtitle 0x2 is 5286510
[09:57:49] reader: done. 1 scr changes
[09:57:50] qsv_enc_init: using 'hardware (1) via D3D11' implementation, API: 1.25
[09:57:50] work: average encoding speed for job is 0.000000 fps
[09:57:50] vfr: 0 frames output, 0 dropped and 0 duped for CFR/PFR
[09:57:50] vfr: lost time: 0 (0 frames)
[09:57:50] vfr: gained time: 0 (0 frames) (0 not accounted for)
[09:57:50] aac-decoder done: 89364 frames, 0 decoder errors
[09:57:50] hevc_qsv-decoder done: 0 frames, 0 decoder errors
[09:57:50] sync: got 0 frames, 91424 expected
[09:57:50] mux: track 0, 0 frames, 0 bytes, 0.00 kbps, fifo 8
[09:57:50] mux: track 1, 190644 frames, 29300591 bytes, 61.48 kbps, fifo 262144
[09:57:50] mux: track 2, 731 frames, 43787 bytes, 0.09 kbps, fifo 1024
[09:57:50] libhb: work result = 0
# Job completed (Elapsed Time: 2m 17s)
# Queue completed

**Comment by RandomEngy** [comment]
I added the ability to do QSV decoding for non-QSV encoders in 3.2 Beta.
